Item 7350: Remove gwtsrc PropertiesEditor and related components

cnathe opened the pull request:
#### Rationale
With the last of the domain designer conversion complete, we are ready to start deleting some GWT PropertiesEditor code and components related to it! There was one final usage of the PropertiesEditor components that needed a little refactor (the "create GPAT assay design from file" scenario). That refactor is included in this PR along with converting the last couple GXT component usages to GWT components.

#### Related Pull Requests
* https://github.com/LabKey/testAutomation/pull/340
* https://github.com/LabKey/provenance/pull/23

#### Changes
* Remove PropertiesEditor and related components from gwtsrc
* Remove PropertiesEditor component usage from DomainImportGrid scenario for assay create from file
* Replace GXT component usages in AssayImporter and DomainImportGrid with GWT components
* Remove GXT dependency from platform/api/build.gradle

cnathe committed:
Remove PropertiesEditor component usage from DomainImportGrid scenario for assay create from file

cnathe committed:
Remove PropertiesEditor and related components from gwtsrc

cnathe committed:
Remove LookupService and related items

cnathe committed:
Remove gwtsrc AssayService

cnathe committed:
Replace GXT component usages in AssayImporter and DomainImportGrid with GWT components

cnathe committed:
DomainImportGrid styling fix for column type info

cnathe committed:
Remove GXT dependency from platform/api/build.gradle

cnathe committed:
Use assay design protocol container when clicking "Show Assay Designer" from assay domain import page

cnathe committed:
Merge branch 'develop' into fb_gwtPropertiesEditorRemoval

labkey-jeckels commented on api/src/org/labkey/api/util/HelpTopic.java:
```diff
@@ -33,6 +32,12 @@
 {
+    // @JavaRuntimeVersion
+    // Update the below constant whenever we add support for a new major Java version so we always point at the current docs.
+    // Why is this global constant defined in such an obscure class? Ideally, we would define it in HelpTopic, but that class
```
Comment is no longer relevant now that it's in a good place.

cnathe replied:
Haha, oops. I guess I copied the constant over like the comment mentioned but didn't think to also update the comment. Done.

labkey-jeckels reviewed (approved):
One small comment fixup suggestion, but in so many ways this is a near-perfect PR!

cnathe committed:
HelpTopic update to comment related to JDK_JAVADOC_BASE_URL constant

cnathe committed:
Bring back gwt AssayService as it was used in provenance module unit test
- but rename it to AssayDomainService so we no longer have two AssayService classes
